packages/material-ui/src/styles/experimentalStyled.test.js

```diff
@@ -52,6 +52,31 @@ describe('experimentalStyled', () => {
     });
   });
 
+  describe('dynamic styles', () => {
+    /**
+     * @type {ReturnType<typeof styled>}
+     */
+    let Div;
+
+    before(() => {
+      // FIXME: Should not error in DEV
+      expect(() => {
+        Div = styled('div')`
+          font-size: ${(props) => props.scale * 8}px;
+          padding: ${(props) => props.scale * 2}px;
+        `;
+      }).toErrorDev(['You have illegal escape sequence in your template literal']);
+    });
+
+    it('can adapt styles to props', () => {
+      render(<Div scale={4} data-testid="target" />);
+      expect(screen.getByTestId('target')).toHaveComputedStyle({
+        fontSize: '32px',
+        padding: '8px',
```

Comment:
Looks like [firefox is computing shorthand properties differently](https://app.circleci.com/pipelines/github/mui-org/material-ui/25266/workflows/bd84898b-bd9d-4d80-9906-01bbc60b810b/jobs/190555). Using longhand properties e.g. `padding-left` instead should work around that.

Reply:
Good finding will refactor the test